### PR TITLE
Delete resource with Access Grant

### PR DESCRIPTION
**Pull request by NSeydoux**
Wrap `@inrupt/solid-client`'s `deleteSolidDataset` and `deleteFile` into the access-grant `resource` module to handle UMA authentication.

# Checklist

- [X] All acceptance criteria are met.
- [X] Relevant documentation, if any, has been written/updated.
- [X] The changelog has been updated, if applicable.
- [X] New functions/types have been exported in `index.ts`, if applicable.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/resource/deleteFile.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+export async function deleteFile(
+  fileUrl: UrlString,
+  accessGrant: VerifiableCredential,
+  options?: FetchOptions
+) {
+  const fetchOptions: FetchOptions = {};
+  if (options && options.fetch) {
+    fetchOptions.fetch = options.fetch;
+  }
```
Comment (jeswr) [inline src/resource/deleteFile.ts:50]:
```suggestion
```

File: src/resource/deleteSolidDataset.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ */
+export async function deleteSolidDataset(
+  datasetUrl: UrlString,
+  accessGrant: VerifiableCredential,
+  options?: FetchOptions
+) {
+  const fetchOptions: FetchOptions = {};
+  if (options && options.fetch) {
+    fetchOptions.fetch = options.fetch;
+  }
```
Comment (jeswr) [inline src/resource/deleteSolidDataset.ts:50]:
```suggestion
```

File: src/resource/deleteSolidDataset.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  const authenticatedFetch = await fetchWithVc(
+    datasetUrl,
+    accessGrant,
+    fetchOptions
```
Comment (jeswr) [inline src/resource/deleteSolidDataset.ts:55]:
```suggestion
    { fetch: options?.fetch }
```

**Review by jeswr — APPROVED** [review]
LGTM - minor stylistic commnts

**Comment by NSeydoux** [comment]
I don't mind the change you are suggesting, but it decreases the code coverage and I don't really have time to look into it, so I'm going to go ahead and not include them.

**Pull request merged**